### PR TITLE
DM-14236: Incorporate l1dbproto into AssociationTask

### PR DESCRIPTION
**Pull request by morriscb**

### CODE REVIEW

File: python/lsst/l1dbproto/l1dbschema.py [hunk 1]
```diff
@@ -262,11 +309,25 @@ def getAfwSchema(self, table_name, columns=None):
...
-                                      units=units,
-                                      parse_strict="silent")
+                if self._afw_type_map_reverse[column.type] == 'String':
+                    key = schema.addField(afw_col,
+                                          type=self._afw_type_map_reverse[column.type],
+                                          doc=column.description or "",
+                                          units=units,
+                                          parse_strict="silent",
+                                          size=10)
+                elif afw_col == 'coord_ra' or afw_col == 'coord_dec':
```
Comment (andy-slac) [inline python/lsst/l1dbproto/l1dbschema.py]: I'm not happy about this, hardcoding column names at this level does not feel right. Can we instead use schema properties to do that? If we want all angles to be Angles in afw.table I would suggest to replace this line with:

    elif units == "deg":

Alternatively we can look at "ucd" property if we want more specific transformations.

But in general we need to think about more general type conversion model sometime later, database schema contain other angle-related combination of units like "deg^2", I'm not sure what should we do for those when we convert between afw and database.

File: python/lsst/l1dbproto/l1dbschema.py [hunk 2]
```diff
@@ -262,11 +309,25 @@ def getAfwSchema(self, table_name, columns=None):
...
+                    key = schema.addField(afw_col,
+                                          type='Angle',
+                                          doc=column.description or "",
+                                          units=units,
```
Comment (andy-slac) [inline python/lsst/l1dbproto/l1dbschema.py]: Should it be "rad" for units, whould it confuse afw.table if you leave "deg" here?

Comment (morriscb) [reply python/lsst/l1dbproto/l1dbschema.py]: It doesn't look like SourceTable cares what you set for units in the key. The type seems to always display as radians until you ask for Angle.asDegrees(). When you request the column an lsst.geom.Angle object is always returned. I can remove the unit from this key definition for clarity if that satisfies.

File: python/lsst/l1dbproto/l1dbschema.py (hunk 1 above)
Comment (morriscb) [reply python/lsst/l1dbproto/l1dbschema.py]: I've changed the test to what you have suggested.

Yeah, for now the code that would compute covariances and produce deg^2 units doesn't exist in the stack yet. I think we'll be needing it soon both for the association work and other locations. I think at that point we'll decide on what object type we store a "deg^2" unit in.

File: python/lsst/l1dbproto/l1dbschema.py (hunk 2 above)
Comment (andy-slac) [reply python/lsst/l1dbproto/l1dbschema.py]: I think it's better to remove it then.

**Pull request merged**